pkg/querier/blocks_store_queryable.go

```diff
@@ -131,6 +132,13 @@ func NewBlocksStoreQueryableFromConfig(querierCfg Config, gatewayCfg storegatewa
 		return nil, errors.Wrap(err, "failed to create bucket client")
 	}
 
+	// Blocks scanner doesn't use chunks, but we pass config for consistency.
+	cachingBucket, err := cortex_tsdb.CreateCachingBucket(storageCfg.BucketStore.ChunksCache, storageCfg.BucketStore.MetadataCache, bucketClient, logger, extprom.WrapRegistererWith(prometheus.Labels{"component": "querier"}, reg))
```

Comment:
Metrics registration LGTM, considering that https://github.com/cortexproject/cortex/pull/2627 will uniform it for the store-gateway too.